### PR TITLE
NewPurchaseMethodDialog: convert to JavaFX

### PR DESCRIPTION
**Pull request by grimreaper**

### CODE REVIEW

File: code/src/java/pcgen/gui3/dialog/NewPurchaseMethodDialogController.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	private Slider pointsEdit;
+	@FXML
+	private Label shownCount;
+
+	@FXML
+	void initialize()
+	{
+		model.nameProperty().bind(nameEdit.textProperty());
+		model.pointsProperty().bind(pointsEdit.valueProperty());
+//		model.pointsProperty().addListener((observable, oldValue, newValue) ->
```
Comment (thpr) [inline code/src/java/pcgen/gui3/dialog/NewPurchaseMethodDialogController.java]: Commented out in new class?

File: code/src/java/pcgen/gui3/dialog/NewPurchaseMethodDialogController.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import javafx.fxml.FXML;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Label;
+import javafx.scene.control.Slider;
+import javafx.scene.control.TextField;
+import javafx.stage.Window;
+
+// consider using Dialog rather than implementing this ourselves
+public class NewPurchaseMethodDialogController
```
Comment (thpr) [inline code/src/java/pcgen/gui3/dialog/NewPurchaseMethodDialogController.java]: Docs for class please

File: code/src/java/pcgen/gui3/dialog/NewPurchaseMethodModel.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ */
+
+package pcgen.gui3.dialog;
+
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.property.SimpleStringProperty;
+import javafx.beans.property.StringProperty;
+
+public class NewPurchaseMethodModel
```
Comment (thpr) [inline code/src/java/pcgen/gui3/dialog/NewPurchaseMethodModel.java]: Docs for class please

File: code/src/java/pcgen/gui3/dialog/NewPurchaseMethodDialogController.java (hunk 1 above)
Comment (grimreaper) [reply code/src/java/pcgen/gui3/dialog/NewPurchaseMethodDialogController.java]: replaced with a direct binding in FXML. I'm not entirely sure which way I like better, but it works.

**Pull request merged**